File: FrontEnd/src/models/Administrador.ts
import Usuario from "./Usuario.ts";
import Telefono from "./Telefono.ts";
import Domicilio from "./Domicilio.ts";

export default class Administrador {
    id?: number;
    nombre: string = "";
    apellido: string = "";
    telefono: Telefono = new Telefono();
    domicilio: Domicilio = new Domicilio();
    usuario: Usuario = new Usuario();
    eliminado: boolean = false;
}

File: BackEnd/src/main/java/com/tienda_ropa/ecommerce/model/Administrador.java
package com.tienda_ropa.ecommerce.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "administrador")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Administrador extends Master {

    private String nombre;
    private String apellido;

    @ManyToOne
    @JoinColumn(name = "id_telefono", nullable = false)
    private Telefono telefono;

    @ManyToOne
    @JoinColumn(name = "id_domicilio", nullable = false)
    private Domicilio domicilio;

    @OneToOne
    @JoinColumn(name = "id_usuario", unique = true)
    @JsonManagedReference
    private Usuario usuarioAdministrador;
}
